docs: add bitbucket samples (#99)
renovatebot/docker-renovate: docs/bitbucket.md

# Samples for Bitbucket pipelines

These are configuration examples for running a self-hosted Renovate on bitbucket.org pipelines.

### bitbucket-pipelines.yml
```yml
image: renovate/renovate:23.96.2-slim

definitions:
  caches:
    renovate: renovate

pipelines:
  default:
    - step:
        name: renovate dry-run
        services:
          - docker
        caches:
          - docker
        script:
          - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/config.js"
          - renovate --dry-run
  branches:
    master:
      - step:
          name: renovate
          services:
            - docker
          caches:
            - docker
          script:
            - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/config.js"
            - renovate
```

### config.js
Example of manual repository config:
```js
module.exports = {
  platform: 'bitbucket',
  username: process.env.USERNAME,
  password: process.env.PASSWORD,
  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
  repositories: [ "user1/repo1", "orh/repo2" ],
}
```

> Using autodiscover

Reviewer: ```suggestion
Using autodiscover:
```

> # Samples for bitbucket pipelines

Reviewer: ```suggestion
# Samples for Bitbucket pipelines
```

> This are samples for running a self-hosted renovate on bitbucket.org pipelines.

Reviewer: ```suggestion
These are configuration examples for running a self-hosted Renovate on bitbucket.org pipelines.
```

> The branches

Reviewer: > The branches

I think this needs more text?

Author: 🤦‍♂️ Don't remember what i want to write, so removing it
```suggestion
```

### bitbucket-pipelines.yml
```yml
image: renovate/renovate:23.96.2-slim

definitions:
  caches:
    renovate: renovate

pipelines:
  default:
    - step:
        name: renovate dry-run
        services:
          - docker
        caches:
          - docker
        script:
          - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/config.js"
          - renovate --dry-run
  branches:
    master:
      - step:
          name: renovate
          services:
            - docker
          caches:
            - docker
          script:
            - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/config.js"
            - renovate
```

### config.js

> Manual repository config

Reviewer: ```suggestion
Example of manual repository config:
```

```js
module.exports = {
  platform: 'bitbucket',
  username: process.env.USERNAME,
  password: process.env.PASSWORD,
  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
  repositories: [ "user1/repo1", "orh/repo2" ],
}
```

Using autodiscover
```js
module.exports = {
  platform: 'bitbucket',
  username: process.env.USERNAME,
  password: process.env.PASSWORD,
  baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
  autodiscover: true,
}
```

### renovate.json

> Use this for self-update renovate

Reviewer: ```suggestion
Use this for self-update Renovate:
```

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [ "config:base" ],
   "regexManagers": [
    {
      "fileMatch": ["^bitbucket-pipelines.yml$"],
      "matchStrings": [
        "image: (?<depName>[a-z/-]+)(?::(?<currentValue>[a-z0-9.-]+))?(?:@(?<currentDigest>sha256:[a-f0-9]+))?"
      ],
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker"
    }
  ]
}
```

### Variables

> You need to defined pipeline variables

Reviewer: ```suggestion
You need to define pipeline variables:
```

- `USERNAME`: Bitbucket.org username
- `PASSWORD`: Bitbucket.org password

> - `GITHUB_COM_TOKEN`: github token to fetch changelog (optional, highly recommend)

Reviewer: ```suggestion
- `GITHUB_COM_TOKEN`: GitHub token to fetch changelog (optional, highly recommended)
```
